### DISCUSSION TITLE
PiecesOS OpenAI compatible

### DISCUSSION THREAD

**Discussion by hra42** [pieces-app/support#382] [Polls]
I'd like to suggest that the piecesOS API should have the same endpoints as the OpenAI API, so we can leverage existing solutions with piecesOS.

**Comment by caleb-at-pieces** [comment]
I'd be interested in what @mark-at-pieces , @sam-at-pieces , @tsavo-at-pieces , @mack-at-pieces have to say about this.

If I have anything to add, I'm not sure if it's our goal to provide generic LLM endpoints to our users, but to be more focused on specific developer based interactions.

**Reply by hra42** [reply]
But to have specific needs you need to do a lot of development work and never won't cover all usecases, maybe some people prefer the way pieces handles the wpe but want to leverage existing tools developed by themselves.

From my pov that should be fairly easy to integrate and make a huge difference in adoption from users.

**Reply by caleb-at-pieces** [reply]
Definitely makes sense here, and adding in Open AI compatibility would certainly help with that.

Another way that this can be done today, albeit a little bit more difficult, would be to utilize one of our SDK's that are open source.

We have some example projects out there on how to call Pieces OS api's utilizing these SDKs, here's a link to some of them:

- python, https://github.com/pieces-app/pieces-os-client-sdk-for-python
- typescript, https://github.com/pieces-app/pieces-os-client-sdk-for-typescript

**Reply by hra42** [reply]
Thanks, I know that there are SDKs the root issue is that I can't put the pOS URL into any URL and have my app ready. The SDKs are good if you start from scratch anyways, really good work from you guys! 🔥

**Comment by tsavo-at-pieces** [comment]
I think this is definitely a no-brainer... Absolutely can do this @hra42 ! Great thoughts here.

**Comment by rosie-at-pieces** [comment]
[#1299](https://github.com/pieces-app/os_server/issues/1299)
